### ISSUE TITLE
[Issue]: 7900xtx system hang - full lock

### ISSUE THREAD

**Issue by terryrankine** [RadeonOpenCompute/ROCm#2689]
### Problem Description

start the machine up
start stable diffusion 
wait until the gpu gets used by chrome and pytorch
[recent.errors.txt](https://github.com/RadeonOpenCompute/ROCm/files/13573816/recent.errors.txt)

```Dec  2 22:11:34 theblob kernel: [ 2649.203914] [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=3
Dec  2 22:11:34 theblob kernel: [ 2649.204158] amdgpu: failed to remove hardware queue from MES, doorbell=0x1002
Dec  2 22:11:34 theblob kernel: [ 2649.204160] amdgpu: MES might be in unrecoverable state, issue a GPU reset
Dec  2 22:11:34 theblob kernel: [ 2649.204164] amdgpu: Failed to evict queue 1
Dec  2 22:11:34 theblob kernel: [ 2649.204166] amdgpu: Failed to evict process queues
Dec  2 22:11:34 theblob kernel: [ 2649.204195] amdgpu 0000:2f:00.0: amdgpu: GPU reset begin!
Dec  2 22:11:35 theblob kernel: [ 2650.229025] amdgpu 0000:2f:00.0: amdgpu: IP block:gfx_v11_0 is hung!
```

after using `sudo rocm-smi --gpureset -d 0` sometimes it works again - but hangs x/wayland

only real fix it cold power off

### Operating System

22.04.3 LTS (Jammy Jellyfish)

### CPU

AMD Ryzen 5 3600X 6-Core Processor

### GPU

Radeon RX 7900 XTX   

### ROCm Version

5.7.2

### Output of /opt/rocm/bin/rocminfo --support

```
terryr@theblob:~$ /opt/rocm/bin/rocminfo --support
ROCk module is loaded
=====================    
HSA System Attributes    
=====================    
Runtime Version:         1.1
System Timestamp Freq.:  1000.000000MHz
Sig. Max Wait Duration:  18446744073709551615 (0xFFFFFFFFFFFFFFFF) (timestamp count)
Machine Model:           LARGE                              
System Endianness:       LITTLE                             
Mwaitx:                  DISABLED
DMAbuf Support:          YES

==========               
HSA Agents               
==========               
*******                  
Agent 1                  
*******                  
  Name:                    AMD Ryzen 5 3600X 6-Core Processor 
  Uuid:                    CPU-XX                             
  Marketing Name:          AMD Ryzen 5 3600X 6-Core Processor 
  Vendor Name:             CPU                                
  Feature:                 None specified                     
  Profile:                 FULL_PROFILE                       
  Float Round Mode:        NEAR                               
  Max Queue Number:        0(0x0)                             
  Queue Min Size:          0(0x0)                             
  Queue Max Size:          0(0x0)                             
  Queue Type:              MULTI                              
  Node:                    0                                  
  Device Type:             CPU                                
  Cache Info:              
    L1:                      32768(0x8000) KB                   
  Chip ID:                 0(0x0)                             
  ASIC Revision:           0(0x0)                             
  Cacheline Size:          64(0x40)                           
  Max Clock Freq. (MHz):   3800                               
  BDFID:                   0                                  
  Internal Node ID:        0                                  
  Compute Unit:            12                                 
  SIMDs per CU:            0                                  
  Shader Engines:          0                                  
  Shader Arrs. per Eng.:   0                                  
  WatchPts on Addr. Ranges:1                                  
  Features:                None
  Pool Info:               
    Pool 1                   
      Segment:                 GLOBAL; FLAGS: FINE GRAINED        
      Size:                    98782896(0x5e34eb0) KB             
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       TRUE                               
    Pool 2                   
      Segment:                 GLOBAL; FLAGS: KERNARG, FINE GRAINED
      Size:                    98782896(0x5e34eb0) KB             
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       TRUE                               
    Pool 3                   
      Segment:                 GLOBAL; FLAGS: COARSE GRAINED      
      Size:                    98782896(0x5e34eb0) KB             
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       TRUE                               
  ISA Info:                
*******                  
Agent 2                  
*******                  
  Name:                    gfx1100                            
  Uuid:                    GPU-f4b3a217836b48ac               
  Marketing Name:          Radeon RX 7900 XTX                 
  Vendor Name:             AMD                                
  Feature:                 KERNEL_DISPATCH                    
  Profile:                 BASE_PROFILE                       
  Float Round Mode:        NEAR                               
  Max Queue Number:        128(0x80)                          
  Queue Min Size:          64(0x40)                           
  Queue Max Size:          131072(0x20000)                    
  Queue Type:              MULTI                              
  Node:                    1                                  
  Device Type:             GPU                                
  Cache Info:              
    L1:                      32(0x20) KB                        
    L2:                      6144(0x1800) KB                    
    L3:                      98304(0x18000) KB                  
  Chip ID:                 29772(0x744c)                      
  ASIC Revision:           0(0x0)                             
  Cacheline Size:          64(0x40)                           
  Max Clock Freq. (MHz):   2482                               
  BDFID:                   12032                              
  Internal Node ID:        1                                  
  Compute Unit:            96                                 
  SIMDs per CU:            2                                  
  Shader Engines:          6                                  
  Shader Arrs. per Eng.:   2                                  
  WatchPts on Addr. Ranges:4                                  
  Features:                KERNEL_DISPATCH 
  Fast F16 Operation:      TRUE                               
  Wavefront Size:          32(0x20)                           
  Workgroup Max Size:      1024(0x400)                        
  Workgroup Max Size per Dimension:
    x                        1024(0x400)                        
    y                        1024(0x400)                        
    z                        1024(0x400)                        
  Max Waves Per CU:        32(0x20)                           
  Max Work-item Per CU:    1024(0x400)                        
  Grid Max Size:           4294967295(0xffffffff)             
  Grid Max Size per Dimension:
    x                        4294967295(0xffffffff)             
    y                        4294967295(0xffffffff)             
    z                        4294967295(0xffffffff)             
  Max fbarriers/Workgrp:   32                                 
  Packet Processor uCode:: 550                                
  SDMA engine uCode::      19                                 
  IOMMU Support::          None                               
  Pool Info:               
    Pool 1                   
      Segment:                 GLOBAL; FLAGS: COARSE GRAINED      
      Size:                    25149440(0x17fc000) KB             
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       FALSE                              
    Pool 2                   
      Segment:                 GLOBAL; FLAGS:                     
      Size:                    25149440(0x17fc000) KB             
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       FALSE                              
    Pool 3                   
      Segment:                 GROUP                              
      Size:                    64(0x40) KB                        
      Allocatable:             FALSE                              
      Alloc Granule:           0KB                                
      Alloc Alignment:         0KB                                
      Accessible by all:       FALSE                              
  ISA Info:                
    ISA 1                    
      Name:                    amdgcn-amd-amdhsa--gfx1100         
      Machine Models:          HSA_MACHINE_MODEL_LARGE            
      Profiles:                HSA_PROFILE_BASE                   
      Default Rounding Mode:   NEAR                               
      Default Rounding Mode:   NEAR                               
      Fast f16:                TRUE                               
      Workgroup Max Size:      1024(0x400)                        
      Workgroup Max Size per Dimension:
        x                        1024(0x400)                        
        y                        1024(0x400)                        
        z                        1024(0x400)                        
      Grid Max Size:           4294967295(0xffffffff)             
      Grid Max Size per Dimension:
        x                        4294967295(0xffffffff)             
        y                        4294967295(0xffffffff)             
        z                        4294967295(0xffffffff)             
      FBarrier Max Size:       32                                 
*** Done ***   
```

**Comment by VeryUnstableRice** [comment]
same card, same problem, same errors. It is not happening only to stable diffusion, but to torch processes that take too much VRAM. If I train my models with 10-14gb the crash does not happen. I assume what's happening is due to the card getting too hot, trying to slow down and the driver mistakes it for a card crash and restarts itself. Or maybe it is due to my motherboard(asus x570 series, I might change it at the recommendation of any rocm dev).

**Comment by VeryUnstableRice** [comment]
https://gitlab.freedesktop.org/agd5f/linux/-/tree/drm-fixes-6.7?ref_type=heads

**Comment by VeryUnstableRice** [comment]
the crash still happens but it's less often. Can you try running something that would normally crash you card in tty, with everything wayland or xorg related killed? I have a suspicion that xwayland is at play here, but I'm not sure

**Comment by VeryUnstableRice** [comment]
did you do what I suggested and increase the fan speed?

**Comment by drhawktopus** [comment]
Possibly useful data point for this family: we've found a deterministic two-context trigger on gfx1100 (Windows, TheRock nightlies) — a second process starting generation work mid-render poisons one context within ~90s (100% busy, ~2% throughput, no TDR/WHEA). The state is context-held and fully recoverable by killing the poisoned process — no reboot, no driver restart. Reproduced 3-for-3 across two runtime versions (7.14 and 7.15 nightlies); 125+ serialized control cycles produced zero occurrences. Full repro steps, negative controls, and telemetry on offer in our report: https://github.com/ROCm/TheRock/issues/6824
